### PR TITLE
Fix locking while writing backup in multiple threads

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Fix locking while writing backup in multiple threads

### CODE REVIEW

File: src/Backups/BackupImpl.cpp [hunk 1]
```diff
@@ -838,6 +841,7 @@ void BackupImpl::writeFile(const String & file_name, BackupEntryPtr entry)
         if (use_archives)
         {
             LOG_TRACE(log, "Adding file {} to archive", adjusted_path);
+            std::lock_guard lock{mutex};
```
Comment (alesapin) [inline src/Backups/BackupImpl.cpp:844]: We still will copy data with this lock to archive. Is it intended?

Comment (vitlibar) [reply src/Backups/BackupImpl.cpp:844]: It's easier to implement it this way here, and writing to an archive is processed strictly in one thread anyway (see the function `supportsWritingInMultipleThreads`)

**Review by alesapin — APPROVED** [review]

**Comment by alesapin** [comment]
Let's wait for tests with sanitizers

**Comment by vitlibar** [comment]
[test_concurrent_backups_s3/test.py::test_concurrent_backups](https://s3.amazonaws.com/clickhouse-test-reports/40819/10b456ce64fcd5e636d4babd0afe424d69173d63/integration_tests__asan__[1/3].html):

2022-09-01 01:07:59 [ 316 ] INFO : Got exception pulling images: Command '['docker-compose', '--env-file', '/ClickHouse/tests/integration/test_concurrent_backups_s3/_instances_gw1_0/.env', '--project-name', 'roottestconcurrentbackupss3', '--file', '/ClickHouse/tests/integration/test_concurrent_backups_s3/_instances_gw1_0/node/docker-compose.yml', '--file', '/compose/docker_compose_minio.yml', 'pull']' timed out after 300 seconds (cluster.py:2341, start)

**Pull request merged**